src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c

```diff
@@ -20,22 +22,154 @@ const ActorInit Bg_Lotus_InitVars = {
     (ActorFunc)BgLotus_Init,
     (ActorFunc)BgLotus_Destroy,
     (ActorFunc)BgLotus_Update,
-    (ActorFunc)BgLotus_Draw
+    (ActorFunc)BgLotus_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Init.s")
+s32 D_80AD6D10[] = { 0x48580064 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Destroy.s")
+void BgLotus_Init(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/func_80AD6830.s")
+    s32 pad;
+    s32 sp2C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/func_80AD68DC.s")
+    Actor_ProcessInitChain(&this->dyna.actor, (InitChainEntry*)D_80AD6D10);
+    DynaPolyActor_Init((DynaPolyActor*)this, 1);
+    DynaPolyActor_LoadMesh(play, (DynaPolyActor*)this, &gLilyPadCol);
```

Comment:
```suggestion
    DynaPolyActor_Init(&this->dyna, DYNA_TRANSFORM_POS);
    DynaPolyActor_LoadMesh(play, &this->dyna, &gLilyPadCol);
```
While casting the overlay struct to dyna works because the dyna is the first element of the struct, and therefor a pointer to the struct, which is a pointer to the start of the struct, is also a pointer to the start of the dyna, the function wants the dyna itself so this is how it should be written.